src/coupled_driver.cpp

```diff
@@ -248,17 +252,36 @@ void CoupledDriver::execute()
 
 double CoupledDriver::temperature_norm(Norm norm)
 {
-  switch (norm) {
-  case Norm::L1: {
-    return xt::norm_l1(temperatures_ - temperatures_prev_)();
-  }
-  case Norm::L2: {
-    return xt::norm_l2(temperatures_ - temperatures_prev_)();
-  }
-  default: {
-    return xt::norm_linf(temperatures_ - temperatures_prev_)();
-  }
+  auto& heat = this->get_heat_driver();
+  double tnorm = 0;
+
+  // Update global temperature norm
+  if (heat.active()) {
+    switch (norm) {
+    case Norm::L1: {
+      double l_sum =
+        xt::eval(xt::sum(xt::abs(cell_temperatures_ - cell_temperatures_prev_)))[0];
```

Comment:
We can't directly use `xt::norm_l2`, since we need to do an MPI collective sum and then take the sqrt of the sum.   

But it does seem like we can `xt::norm_l1` and then do the MPI collective sum.  Likewise, I think we can use `xt::norm_linf` and do then do the MPI collective max.

Reply:
Re: the L2 norm, I just discovered [`xt::norm_sq`](https://xtensor.readthedocs.io/en/latest/api/reducing_functions.html#_CPPv4I0000EN2xt7norm_sqEDaRR1ERR1X3EVS).  With that, I think we could do something like:
```
case Norm::L2: {
   double l_norm_sq = xt::norm_sq(cell_temperatures_ - cell_temperatures_prev_);
   double g_norm_sq;
   MPI_Reduce(&l_norm_sq, &g_norm_sq, 1, MPI_DOUBLE, MPI_SUM, 0, heat.comm_.comm);
   tnorm = std::sqrt(g_norm_sq);
   break;
}
```

Reply:
Fixed in 3bbdc72